scripts/backCompatChecker.py

```diff
@@ -9,6 +9,7 @@
 
 oldArguments = []
 newArguments = []
+allowedMissingArguments = ['--azure-rm-service-prinicipal-key']
```

Comment:
This is prone to letting issues creep in. 
How would we solve this is the parameter is generic like "project" name? If I ignore that it will be ignored for all commands.

Regarding the PR for 5.0 upgrade. I would rather not want to add exceptions here for long term. We know whatever is breaking in this major upgrade. If I add checks for those here, after one release those checks should not be applied.

Reply:
We can use a dictionary to specifically allow exceptions for params corresponding to each command. e.g. { command1: [arg1, arg2], command2: [arg1, arg3, arg4] }

I can use the same for 5.0 unless you have a better solution in mind. I was thinking of disabling this for one release since we are introducing breaking changes but then this looks like a good way to catch all the breaking changes, document them in release notes and then get rid of the checks next time.